### PR TITLE
Implement `Ordered` interface in ByteVector and BitVector

### PR DESCRIPTION
**Pull request by sideeffffect**
Steps towards https://github.com/scodec/scodec-cats/issues/13

/cc @mpilquist

### CODE REVIEW

**Comment by sideeffffect** [comment]
@mpilquist :pray:

File: core/shared/src/main/scala/scodec/bits/BitVector.scala [hunk 1]
```diff
@@ -1320,6 +1323,30 @@ sealed abstract class BitVector extends BitwiseOperations[BitVector, Long] with
...
+      val commonLength = thisLength.min(thatLength)
+      var i = 0
+      while (i < commonLength) {
+        val cmp = this(i).compare(that(i))
+        if (cmp != 0) {
+          return cmp
+        }
+        i = i + 1
+      }
+      if (thisLength < thatLength) {
```
Comment (mpilquist) [inline core/shared/src/main/scala/scodec/bits/BitVector.scala:1342]: Why prefer comparing elements over length? As opposed to first comparing length and only if equal, comparing elements?

Comment (sideeffffect) [reply core/shared/src/main/scala/scodec/bits/BitVector.scala:1342]: This is supposed to do a [_lexicographical_ ordering](https://en.wikipedia.org/wiki/Lexicographical_order) of the Byte vectors. Lengths are compared _only after_ we have established that one is a prefix of the other.
Before you do that (i.e. you compare their elements), the lengths alone can't tell you which ByteVector is greater.
See for example this test case:
```scala
assert(ByteVector(1, 2) < ByteVector(2))
```
The first one may be longer, but is still lesser that the second one.

**Comment by mpilquist** [comment]
Can you backport this to series/1.1.x too?

**Pull request merged**